x/warp/config.go

```diff
@@ -166,6 +170,10 @@ func (c *Config) PredicateGas(predicateBytes []byte) (uint64, error) {
 	if err != nil {
 		return 0, fmt.Errorf("%w: %s", errInvalidWarpMsg, err)
 	}
+	_, err = payload.Parse(warpMessage.Payload)
+	if err != nil {
+		return 0, fmt.Errorf("%w: %s", errInvalidWarpMsgPayoad, err)
+	}
```

Comment:
The only returned from `IntrinsicGas` is `ErrGasUintOverflow`. I wonder if we should do this verification (and the `err` returned from `ParseMessage`) in elsewhere. I don't think payload has anyhow related with gas calculation. 

I'm a bit in favor of using gas "pessimistically" in almost every case to prevent any possible DoS attacks. Alternative to moving parse verification to elsewhere, we can define a base gas cost for predicates and return that in case of errors other than overflows.

Reply:
This is executed in the tx pool, so it's already run "pessimistically".

This seemed like the most convenient place to put it since it's already used where needed and invalidates the transaction. Where else would you suggest putting it?